### PR TITLE
fix(api): Use internal example in KeyFrameEffect pages

### LINKED ISSUE #25694: KeyframeEffect.target: issues with the example
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/KeyframeEffect/target

### What specific section or headline is this issue about?

examples

### What information was incorrect, unhelpful, or incomplete?

The white rabbit example should be a transition, not an animation: The ears go from visible to not. You don’t need an animation for that, just a class change. 

Let's say we stick with that anyhow, there are other issues. 

The code in the code example:

```javascript
// returns <div id="rabbit">Click the rabbit's ears!</div>
rabbitDownKeyframes.target;
```

Is not included in the linked Codepen.
If the user wants to add it by copying and pasting, the line itself isn’t a log or a variable definition. it just sits there.

Additionally, the example is not good UX. The sighted user has no way of knowing where to click or even that they should. There are no focusable / interactive html elements. The cursor doesn't indicate clickability either.

### What did you expect to see?

in page example. If not, a link to a working example that includes the API component being discussed

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/keyframeeffect/target`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/KeyframeEffect/target
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/keyframeeffect/target/index.md
* Last commit: https://github.com/mdn/content/commit/21d3e89589aaf9e5cfa667de679134513ab833f3
* Document last modified: 2023-02-28T12:02:51.000Z

</details>

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- fixes https://github.com/mdn/content/issues/25694

As asked for in the issue, the PR adds examples in the KeyFrameEffect pages.

The new example makes :rofl: emojy really roll on the floor. Also, `KeyframeEffect.spacing` doesn't exist in specs or MDN so it's been removed.

### CODE REVIEW

**Comment by chrisdavidmills** [comment]
Sorry, I've had so many reviews dumped on me recently that I'm struggling to keep up with them all. I'm removing myself from a couple of them.

**Comment by bsmth** [comment]
Hey Onkar, I've added myself as a reviewer. I'll try to take a look today 🙏🏻

File: files/en-us/web/api/keyframeeffect/pseudoelement/index.md [hunk 1]
```diff
@@ -30,12 +30,14 @@ A string or `null`.
 
 ```css
 #text::after {
-  content: "⭐";
+  content: "👹";
```
Comment (OnkarRuikar) [inline files/en-us/web/api/keyframeeffect/pseudoelement/index.md:33]: The spinning yellow icons have poor contrast with the white background. The spinning is hardly noticeable.

File: files/en-us/web/api/keyframeeffect/pseudoelement/index.md [hunk 2]
```diff
@@ -72,7 +74,7 @@ switchPseudoElement();
 logPseudoElement();
 ```
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/pseudoelement/index.md:75]: question: Is it necessary to have 

```js
[
    { transform: "rotate(0)" },
    { transform: "rotate(180deg)" },
    { transform: "rotate(360deg)" },
],
```

or is this enough?

```js
[
    { transform: "rotate(360deg)" },
],
```

File: files/en-us/web/api/keyframeeffect/index.md [hunk 3]
```diff
@@ -44,29 +44,55 @@ _This interface inherits some of its methods from its parent, {{domxref("Animati
 
 ## Examples
 
-In the [Follow the White Rabbit example](https://codepen.io/rachelnabors/pen/eJyWzm/?editors=0010), the KeyframeEffect constructor is used to create a set of keyframes that dictate how the White Rabbit should animate down the hole:
+In the following example, the KeyframeEffect constructor is used to create a set of keyframes that dictate how the rofl emoji should role on the floor:
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/index.md:47]: :)
```suggestion
In the following example, the KeyframeEffect constructor is used to create a set of keyframes that dictate how the rofl emoji should roll on the floor:
```

File: files/en-us/web/api/keyframeeffect/keyframeeffect/index.md [hunk 4]
```diff
@@ -76,29 +76,53 @@ The single argument constructor (see above) creates a clone of an existing {{dom
 
 ## Examples
 
-In the [Follow the White Rabbit example](https://codepen.io/rachelnabors/pen/eJyWzm/?editors=0010), the `KeyframeEffect` constructor is used to create a set of keyframes that dictate how the White Rabbit should animate down the hole:
+In the following example, the KeyframeEffect constructor is used to create a set of keyframes that dictate how the emoji should role on the floor:
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/keyframeeffect/index.md:79]:
```suggestion
In the following example, the KeyframeEffect constructor is used to create a set of keyframes that dictate how the emoji should roll on the floor:
```

File: files/en-us/web/api/keyframeeffect/getkeyframes/index.md [hunk 5]
```diff
@@ -27,24 +27,65 @@ Returns a sequence of objects with the following format:
...
+    direction: "alternate",
+    easing: "ease-in-out",
+    iterations: "Infinity",
+  },
+);
+
+const rollingAnimation = new Animation(rollingKeyframes, document.timeline);
+rollingAnimation.play();
+
+console.log(rollingAnimation.effect.getKeyframes());
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/getkeyframes/index.md:63]: do you want to show what it logs?

File: files/en-us/web/api/keyframeeffect/getkeyframes/index.md [hunk 6]
```diff
@@ -27,24 +27,65 @@ Returns a sequence of objects with the following format:
...
+```
+
+```html
+<div>🤣</div>
+<hr />
+```
 
-redQueen_alice.effect.getKeyframes();
+```css
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/getkeyframes/index.md:71]: maybe we can hide the CSS (also applies to others) what do you think?

**Comment by bsmth** [comment]
Looking great. I have one concern that we're duplicating the code so next time we make changes, we have to update multiple places, but I'm not sure what the better option would be. 

Do you have any thoughts about simplifying the [KeyframeEffect/target](https://pr34613.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/KeyframeEffect/target) example so it's more minimal? Getting rid of the `<hr>` and hiding the CSS block?

Otherwise, good to go, IMO

File: files/en-us/web/api/keyframeeffect/pseudoelement/index.md (hunk 2 above)
Comment (OnkarRuikar) [reply files/en-us/web/api/keyframeeffect/pseudoelement/index.md:75]: done

File: files/en-us/web/api/keyframeeffect/getkeyframes/index.md (hunk 5 above)
Comment (OnkarRuikar) [reply files/en-us/web/api/keyframeeffect/getkeyframes/index.md:63]: Playground logs `[object Object],[object Object]` and browser consoles don't log text but rather inspectable objects. There seems no useful text representation.

**Comment by OnkarRuikar** [comment]
>  I have one concern that we're duplicating the code so next time we make changes, we have to update multiple places, but I'm not sure what the better option would be.

Unfortunately, all the examples are not the same they have sight variations. Some logs the corresponding member/function in the code.

> Do you have any thoughts about simplifying the [KeyframeEffect/target](https://pr34613.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/KeyframeEffect/target) example so it's more minimal? Getting rid of the `<hr>` and hiding the CSS block?

CSS block is not relevant so I've hidden it. But how can you rofl without a floor?! :rofl:

File: files/en-us/web/api/keyframeeffect/getkeyframes/index.md (hunk 5 above)
Comment (bsmth) [reply files/en-us/web/api/keyframeeffect/getkeyframes/index.md:63]: You could do 

```js
// Array [ {…}, {…} ]
```

File: files/en-us/web/api/keyframeeffect/pseudoelement/index.md [hunk 7]
```diff
@@ -72,7 +71,7 @@ switchPseudoElement();
...
 
-{{EmbedLiveSample("Examples", "100", "70")}}
+{{EmbedLiveSample("Examples", "100", "80")}}
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/pseudoelement/index.md:74]: I get vertical scrollbar with 80, but not with 90:

```suggestion
{{EmbedLiveSample("Examples", "100", "90")}}
```

File: files/en-us/web/api/keyframeeffect/target/index.md [hunk 8]
```diff
@@ -16,21 +16,53 @@ An {{domxref("Element")}} or `null`.
...
+div {
+  width: fit-content;
+  margin-left: calc(50% - 132px);
+  font-size: 64px;
+  user-select: none;
+  margin-top: 1rem;
+}
+```
+
+{{ EmbedLiveSample("Examples", "100%", '120') }}
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/target/index.md:64]: Nit: let's not mix quotes :)

```suggestion
{{ EmbedLiveSample("Examples", "100%", "120") }}
```

File: files/en-us/web/api/keyframeeffect/keyframeeffect/index.md [hunk 9]
```diff
@@ -76,29 +76,50 @@ The single argument constructor (see above) creates a clone of an existing {{dom
...
+div {
+  width: fit-content;
+  margin-left: calc(50% - 132px);
+  font-size: 64px;
+  user-select: none;
+  margin-top: 1rem;
+}
+```
+
+{{ EmbedLiveSample("Examples", "100%", '120') }}
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/keyframeeffect/index.md:121]:
```suggestion
{{ EmbedLiveSample("Examples", "100%", "120") }}
```

File: files/en-us/web/api/keyframeeffect/index.md [hunk 10]
```diff
@@ -44,29 +44,52 @@ _This interface inherits some of its methods from its parent, {{domxref("Animati
...
+div {
+  width: fit-content;
+  margin-left: calc(50% - 132px);
+  font-size: 64px;
+  user-select: none;
+  margin-top: 1rem;
+}
+```
+
+{{ EmbedLiveSample("Examples", "100%", '120') }}
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/index.md:91]:
```suggestion
{{ EmbedLiveSample("Examples", "100%", "120") }}
```

File: files/en-us/web/api/keyframeeffect/getkeyframes/index.md [hunk 11]
```diff
@@ -27,24 +27,62 @@ Returns a sequence of objects with the following format:
...
+div {
+  width: fit-content;
+  margin-left: calc(50% - 132px);
+  font-size: 64px;
+  user-select: none;
+  margin-top: 1rem;
+}
 ```
 
+{{ EmbedLiveSample("Examples", "100%", '120') }}
```
Comment (bsmth) [inline files/en-us/web/api/keyframeeffect/getkeyframes/index.md:84]:
```suggestion
{{ EmbedLiveSample("Examples", "100%", "120") }}
```

**Review by bsmth — APPROVED** [review]
Thanks a lot! Couple of minor suggestions for you, but I think we're ready to go when you've had a look at my comments 👍🏻

File: files/en-us/web/api/keyframeeffect/getkeyframes/index.md (hunk 5 above)
Comment (OnkarRuikar) [reply files/en-us/web/api/keyframeeffect/getkeyframes/index.md:63]:
```suggestion
// Array [ {…}, {…} ]
console.log(rollingAnimation.effect.getKeyframes());
```

**Comment by bsmth** [comment]
Looks great! Merging now!

**Pull request merged**